src/main/java/com/ontotext/ehri/deduplication/model/USHMMClassificationInstance.java

```diff
@@ -0,0 +1,161 @@
+package com.ontotext.ehri.deduplication.model;
+
+import com.ontotext.ehri.deduplication.measures.JaroWinkler;
+import com.ontotext.ehri.deduplication.measures.Levenshtein;
+import com.ontotext.ehri.deduplication.measures.USHMMDate;
+import org.apache.commons.codec.EncoderException;
+import org.apache.commons.codec.language.DaitchMokotoffSoundex;
+import org.apache.commons.codec.language.bm.BeiderMorseEncoder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import types.Alphabet;
+import types.SparseVector;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class USHMMClassificationInstance {
+
+    private Alphabet xA;
+    private SparseVector sparseVector;
+
+    private USHMMPerson person1;
+    private USHMMPerson person2;
+
+    private static final DaitchMokotoffSoundex daitchMokotoffSoundex = new DaitchMokotoffSoundex();
+    private static final BeiderMorseEncoder beiderMorseEncoder = new BeiderMorseEncoder();
+
+    private static final Logger logger = LoggerFactory.getLogger(USHMMPersonsFeatureExtractor.class);
+
+    public USHMMClassificationInstance(Alphabet xA, USHMMPerson person1, USHMMPerson person2) {
+        this.xA = xA;
+        this.sparseVector = new SparseVector();
+
+        this.person1 = person1;
+        this.person2 = person2;
+    }
+
+    public SparseVector getSparseVector() {
+        extractNamesFeatures();
+        extractMotherNameFeatures();
+        extractPlaceBirthFeatures();
+        extractDateBirthFeatures();
+        extractGenderFeature();
+        extractSourceFeature();
+        extractPersonTypeFeature();
+        extractOccupationFeature();
+        return sparseVector;
+    }
+
+    private void extractNamesFeatures() {
+        extractJaroWinklerNamesFeatures();
+        extractJaroWinklerNormalizedNamesFeatures();
+        extractDoubleMetaphoneFeatures();
+        extractBeiderMorseFeatures();
+        extractDaitchMokotoffFeatures();
+    }
+
+    private void extractJaroWinklerNamesFeatures() {
+        sparseVector.add(xA.lookupObject("jwf"), JaroWinkler.distance(person1.firstName, person2.firstName));
+        sparseVector.add(xA.lookupObject("jwl"), JaroWinkler.distance(person1.lastName, person2.lastName));
+
+        sparseVector.add(xA.lookupObject("jw"), JaroWinkler.distance(person1.firstName + " " + person1.lastName,
+                person2.firstName + " " + person2.lastName));
+    }
+
+    private void extractJaroWinklerNormalizedNamesFeatures() {
+        sparseVector.add(xA.lookupObject("jwnf"), JaroWinkler.distance(person1.normalizedFirstName, person2.normalizedFirstName));
```

Comment:
Do you think that just taking the distance is a good feature, maybe discretizing it to bins will do a better job for classification?